lib/json_api_controller/updatable_resource.rb

```diff
@@ -29,11 +29,10 @@ def update_links
       resource = controlled_resources.first
       resource_class.transaction(requires_new: true) do
         add_relation(resource, relation, params[relation])
-        if resource.changed?
-          resource.save!
-        else
-          resource.touch
-        end
+        # as this resource may not have changed but the linked resources may have
+        # ensure we modify the upated_at timestamp to cache bust this resource
+        resource.updated_at = Time.zone.now
+        resource.save!
```

Comment:
I thought #touch was just sugar for `resource.update_attributes(:updated_at, Time.zone.now)` or basically what this looks like it does. Does this do something different?

Reply:
`.touch` ran into issues with a callback where the resource's owner could not be read. 
tbh (non sequitur) i have found touch to be finnicky since rails 5 (maybe even before). https://github.com/rails/rails/issues/34768, we see this issue a bit in SubjectSetSubjectCounterWorker

Reply:
the original issue what the `.save!` call was not running as the `.changed?` returned false. That conditional was added to avoid 2 round trips to the db on each `update_links` action, i.e.
- one if if we need to save and `updated_at` will change and 
- another if the resource hasn't changed but we want to cache bust via touching `updated_at`

I couldn't see an API interface to detect the the associated records may have changed and instead of running both the save and the touch (2 db round trips) i figured manually modifying the `updated_at` on the original resource was a quick fix for both use cases.

FWIW i don't think `.touch` if finicky and we don't use `counter_caches` on our records, rather we're hitting long running counter jobs and the resource changes under us. If that's the case we should reload them before saving where these stale object errors surface.